src/components/Page/Toolbar/Toolbar.jsx

```diff
@@ -40,53 +42,61 @@ export class Toolbar extends Component {
     this._initializeModal();
   }
 
-  onBrushSizeChange = () => {
-    this.props.onBrushSizeChange(Number(this.brushSizeRangeRef.current.value));
-    this.setState({
-      brushSize: Number(this.brushSizeRangeRef.current.value)
-    })
-  }
-
-  onEraserSizeChange = () => {
-    this.props.onEraserSizeChange(Number(this.eraserSizeRangeRef.current.value));
-    this.setState({
-      eraserSize: Number(this.eraserSizeRangeRef.current.value)
-    })
+  provideRef = (tool) => {
+    if(tool === 'brush')
+      return this.brushSizeRangeRef.current.value;
+    else
+    if(tool === 'rainbow_brush')
+      return this.changeRateRangeRef.current.value;
+    else
+    if(tool === 'eraser')
+      return this.eraserSizeRangeRef.current.value;
+    else
+    if(tool === 'line')
+      return this.lineThicknessRangeRef.current.value;
   }
 
-  onLineSizeChange = () => {
-    this.props.onBrushSizeChange(Number(this.lineSizeRangeRef.current.value));
+  manageChange = (property,tool = 'brush') => {
+    let result = this.provideRef(tool);
+    this.props._changeToolSetting(property,Number(result));
     this.setState({
-      brushSize: Number(this.lineSizeRangeRef.current.value)
-    })
+      property: Number(this.colorSpeedRangeRef.current.value)
```

Comment:
This will set the value property named `property` literally. You have to do `[property]: Number(...)`

Reply:
ok now i got this thanks 😊